### DISCUSSION TITLE
Teampass 3.1.1

### DISCUSSION THREAD

**Discussion by nilsteampassnet** [nilsteampassnet/TeamPass#4060] [General]
## What's new
This release embeds next fixes:

- Improved upgrade from V2
- Improved upgrade process
- Improved attachment download
- Optimized background tasks handler
- Fixed and improved deletion utility
- Added minimal edition lock delay for an item
- Fixed issue in deleting folder if user has his cache table not updated
- Fixed issue in search page, clicking on pwd copying icon doesn't work
- Fixed issue in search filter
- Fixed number of users online
- Fixed issue in OTV
- Fixes for #4005, #4006, #4016, #4018, #3968, #4021, #4014, #3986, #4048, #4049, #4056, #4058
- Security: removed all usage of $_GET, $_POST and $_SERVER variables
- Full list of changes: [3.1.0 -> 3.1.1](https://github.com/nilsteampassnet/TeamPass/compare/3.1.0...3.1.1?expand=1)

## Important
- Requires at least `PHP 8.1`
- New password library implemented, [read about impacts](https://github.com/nilsteampassnet/TeamPass/discussions/4020)

## Languages

Please join [Teampass v3 translation project on Poeditor](https://poeditor.com/projects/view?id=433631) and translate it for your language.

## Installation

Follow instructions from [Documentation](https://documentation.teampass.net/#/install/installation).

## Upgrade

Follow instructions from [Documentation](https://documentation.teampass.net/#/install/upgrade).

## Ideas and comments

Are welcome ... please use [Discussions](https://github.com/nilsteampassnet/TeamPass/discussions).

<hr /><em>This discussion was created from the release <a href='https://github.com/nilsteampassnet/TeamPass/releases/tag/3.1.1'>Teampass 3.1.1</a>.</em>

**Comment by chostik** [comment]
I wanted to ask when you plan to implement "Changing password on distant server" feature?
